catroid/src/main/java/org/catrobat/catroid/ui/Multilingual.java

```diff
@@ -0,0 +1,469 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.ui;
+
+import android.annotation.TargetApi;
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.content.res.Configuration;
+import android.content.res.Resources;
+import android.os.Build;
+import android.os.Bundle;
+import android.support.annotation.Nullable;
+import android.util.DisplayMetrics;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.ListView;
+
+import org.catrobat.catroid.R;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import static org.catrobat.catroid.CatroidApplication.defaultSystemLanguage;
+import static org.catrobat.catroid.CatroidApplication.languageSharedPreferences;
+
+public class Multilingual extends Activity {
+	private static String[] languageCode = {"az", "bs", "ca", "cs", "sr-rCS", "sr-rSP", "da", "de", "en-rAU", "en-rCA",
+			"en-rGB", "en", "es", "fr", "gl", "hr", "in", "it", "hu", "mk", "ms", "nl", "no", "pl", "pt-rBR", "pt", "ru",
+			"ro", "sq", "sl", "sv", "vi", "tr", "ml", "ta", "te", "th", "gu", "hi", "ja", "ko", "zh-rCN", "zh-rTW", "ar",
+			"ur", "fa", "ps", "sd", "iw"};
+	public static String languageTagKey = "applicationLanguage";
+
+	@Override
+	protected void onCreate(@Nullable Bundle savedInstanceState) {
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_multilingual);
+		setTitle(R.string.preference_title_language);
+		ListView listview = (ListView) findViewById(R.id.list_languages);
+		List<String> languagesNames = new ArrayList<>();
+		languagesNames.add(getResources().getString(R.string.device_language));
+		for (String aLanguageCode : languageCode) {
+			if (aLanguageCode.length() == 2) {
+				languagesNames.add(new Locale(aLanguageCode).getDisplayName(new Locale(aLanguageCode)));
+			} else {
+				String language = aLanguageCode.substring(0, 2);
+				String country = aLanguageCode.substring(4);
+				languagesNames.add(new Locale(language, country).getDisplayName(new Locale(language, country)));
```

Comment:
`languagesNames.add(new Locale(language, country).getDisplayName(new Locale(language, country)));
` the output will be the same languageName in its language!
but when you use `languagesNames.add(new Locale(language, country).getDisplayName());
` the name of the language will be translated into the current locale of the App or the smartPhone.
see more [https://developer.android.com/reference/java/util/Locale.html#getDisplayName(java.util.Locale)](url)